src/components/editor/MonacoEditor.tsx

```diff
@@ -119,10 +117,15 @@ function MonacoEditor({
     };
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    const debouncedChange = useCallback(debounce(updateValue, 1000), [
-        currentCatalog,
+    const debouncedChange = useCallback(debounce(updateValue, 750), [
+        currentCatalog.catalog_name,
     ]);
 
+    const specAsString = useMemo(
+        () => stringifyJSON(currentCatalog.spec),
+        [currentCatalog.spec]
+    );
```

Comment:
The stringify isn't a _huge_ calculation but it seems this was messing stuff up for Windows... potentially messed it up in any OS and just saw it on Windows because it is slower due to running in a VM for me.